### ISSUE TITLE
Schema mismatch for read only open async.

### ISSUE THREAD

**Issue by morten-krogh** [realm/realm-object-store#607]
When an app opens a sync Realm with asyncOpen, it receives the initial schema with the 
initial download from the server. 
If the app has a schema with extra columns compared to the server's schema, the app will
add the extra columns and upload them to the server. 
If the permission to the server Realm is read-only, this chain of events lead to a permission dnied error from the server.

A better behavior would be that the client issues a "schema mismatch" error. This behavior would make it easier for the app developer to debug the problem.

In summary: Issue a "schema mismatch" error after async open returns if the Realm is read-only and the app schema contains extra columns.

See the issue 
https://github.com/realm/realm-object-server/issues/259#issuecomment-348319532
for an example.

**Comment by bmunkholm** [comment]
Moved to https://github.com/realm/realm-core/issues/4610
